v2/stream/datav2_test.go

```diff
@@ -0,0 +1,165 @@
+package stream
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/vmihailenco/msgpack/v5"
+)
+
+// tradeWithT is the incoming trade message that also contains the T type key
+type tradeWithT struct {
+	Type       string    `msgpack:"T"`
+	ID         int64     `msgpack:"i"`
+	Symbol     string    `msgpack:"S"`
+	Exchange   string    `msgpack:"x"`
+	Price      float64   `msgpack:"p"`
+	Size       uint32    `msgpack:"s"`
+	Timestamp  time.Time `msgpack:"t"`
+	Conditions []string  `msgpack:"c"`
+	Tape       string    `msgpack:"z"`
+}
+
+// quoteWithT is the incoming quote message that also contains the T type key
+type quoteWithT struct {
+	Type        string    `msgpack:"T"`
+	Symbol      string    `msgpack:"S"`
+	BidExchange string    `msgpack:"bx"`
+	BidPrice    float64   `msgpack:"bp"`
+	BidSize     uint32    `msgpack:"bs"`
+	AskExchange string    `msgpack:"ax"`
+	AskPrice    float64   `msgpack:"ap"`
+	AskSize     uint32    `msgpack:"as"`
+	Timestamp   time.Time `msgpack:"t"`
+	Conditions  []string  `msgpack:"c"`
+	Tape        string    `msgpack:"z"`
+}
+
+// barWithT is the incoming bar message that also contains the T type key
+type barWithT struct {
+	Type      string    `msgpack:"T"`
+	Symbol    string    `msgpack:"S"`
+	Open      float64   `msgpack:"o"`
+	High      float64   `msgpack:"h"`
+	Low       float64   `msgpack:"l"`
+	Close     float64   `msgpack:"c"`
+	Volume    uint64    `msgpack:"v"`
+	Timestamp time.Time `msgpack:"t"`
+}
+
+var testTime = time.Date(2021, 03, 04, 15, 16, 17, 18, time.UTC)
+
+var testTrade = tradeWithT{
+	Type:       "t",
+	ID:         42,
+	Symbol:     "TEST",
+	Exchange:   "X",
+	Price:      100,
+	Size:       10,
+	Timestamp:  testTime,
+	Conditions: []string{" "},
+	Tape:       "A",
+}
+
+var testQuote = quoteWithT{
+	Type:        "q",
+	Symbol:      "TEST",
+	BidExchange: "B",
+	BidPrice:    99.9,
+	BidSize:     100,
+	AskExchange: "A",
+	AskPrice:    100.1,
+	AskSize:     200,
+	Timestamp:   testTime,
+	Conditions:  []string{"R"},
+	Tape:        "B",
+}
+
+var testBar = barWithT{
+	Type:      "b",
+	Symbol:    "TEST",
+	Open:      100,
+	High:      101.2,
+	Low:       98.67,
+	Close:     101.1,
+	Volume:    2560,
+	Timestamp: time.Date(2021, 03, 05, 16, 0, 0, 0, time.UTC),
+}
+
+func TestHandleMessages(t *testing.T) {
```

Comment:
What about adding some unexpected fields to these structs? That would help ensure the API can be extended later on without breaking it for the users of the go API.